### PR TITLE
Google vision api

### PR DESCRIPTION
**Pull request by pingunaut**
## Status
Ready

## Description
This integration add initial support for the connection of the Google Vision API.
As a first use case, it can be utilized to detect brand logos in an image.
But as the integration contains all helper functions to connect to Vision API in general, it can be enhanced in the future to also add support for other Vision API features like image labeling.

## Screenshots

image

## Related PRs
Uses the newly introduced google-vision-api docker image: https://github.com/demisto/dockerfiles/pull/338

## Required version of Demisto
4.5.0

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

### CODE REVIEW

**Comment by Itay4** [comment]
@pingunaut looks good !
did you get to look at my general comments? need our help with that?

**Comment by pingunaut** [comment]
Hi @Itay4, everything clear so far. Was not able to find any time to finish the the open topics, but will do so asap

**Comment by Itay4** [comment]
@pingunaut Sure, no worries 🙂
Let me know if you need any help

**Comment by pingunaut** [comment]
Yes. Will do

**Comment by pingunaut** [comment]
Hi @Itay4 
please have a look again. Worked on all comments, except unit tests... I wasn't able to run them locally pytest was not able to resolve the Demisto dependencies.

**Comment by Itay4** [comment]
@pingunaut looks good, will review soon and contact you on DFIR to schedule execution review session

File: TestPlaybooks/playbook-Google_Vision_API_Test.yml [hunk 1]
```diff
@@ -0,0 +1,244 @@
+id: GoogleVisionAPI-Test
```
Comment (Itay4) [inline TestPlaybooks/playbook-Google_Vision_API_Test.yml]: Google Vision API - Test

File: TestPlaybooks/playbook-Google_Vision_API_Test.yml [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+version: -1
+name: Google Vision API Test
```
Comment (Itay4) [inline TestPlaybooks/playbook-Google_Vision_API_Test.yml]: Google Vision API - Test

File: Integrations/GoogleVisionAPI/GoogleVisionAPI.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      isArray: true
+      name: entry_id
+      required: true
+      secret: false
+    deprecated: false
+    description: Detects brand logos in the given image
+    execution: false
+    name: google-vision-detect-logos
+    outputs:
+    - contextPath: GoogleVisionAPI
```
Comment (Itay4) [inline Integrations/GoogleVisionAPI/GoogleVisionAPI.yml]: I think you are not handling it correctly in the code, let's go over this together in the execution review.

File: TestPlaybooks/playbook-Google_Vision_API_Test.yml [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+    timertriggers: []
+    ignoreworker: false
+  "33":
+    id: "33"
+    taskid: f5780469-d422-4e36-8870-17b726daf3e7
+    type: regular
+    task:
+      id: f5780469-d422-4e36-8870-17b726daf3e7
+      version: -1
+      name: Close Investigation
```
Comment (Itay4) [inline TestPlaybooks/playbook-Google_Vision_API_Test.yml]: This isn't needed, can remove.

**Comment by Itay4** [comment]
@richardbluestone @kirbles19 Please review the YAML and the MD files

File: TestPlaybooks/playbook-Google_Vision_API_Test.yml (hunk 1 above)
Comment (Itay4) [inline TestPlaybooks/playbook-Google_Vision_API_Test.yml]:
```suggestion
id: Google Vision API - Test
```

File: TestPlaybooks/playbook-Google_Vision_API_Test.yml (hunk 2 above)
Comment (Itay4) [inline TestPlaybooks/playbook-Google_Vision_API_Test.yml]:
```suggestion
name: Google Vision API - Test
```

**Comment by kirbles19** [comment]
@Itay4 Done.

**Comment by Itay4** [comment]
@pingunaut i understand you cannot CLA, so I'm closing this for now. 
please reopen if anything changed on your side.
thanks !

**Comment by pingunaut** [comment]
Hi @Itay, signed weeks ago. BR Martin 

Am 6. Januar 2020 08:52:42 MEZ schrieb Itay Keren <notifications@github.com>:
>@pingunaut i understand you cannot CLA, so I'm closing this for now. 
>please reopen if anything changed on your side.
>thanks !
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/demisto/content/pull/4122#issuecomment-571041065

-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Comment by anara123** [comment]
@pingunaut it shows that you haven't signed yet. Can you try to sign it again please
image

**Comment by pingunaut** [comment]
It shows wrong. Cannot do it twice... no more sign button there as it is signed already.
image

**Comment by anara123** [comment]
@pingunaut you are right. I see that you signed the CLA, for some reason in github it shows that you haven't. 
Sorry for misunderstanding. Reopening.
@Itay4 you approve the PR? Can we merge?

**Pull request merged**